client/src/unifycr.c

```diff
@@ -1096,65 +1115,66 @@ int unifycr_fid_open(const char* path, int flags, mode_t mode, int* outfid, off_
 
     int gfid = -1, rc = 0;
     if (fs_type == UNIFYCR_LOG) {
-		if (fid < 0) {
-			rc = unifycr_get_global_fid(path, &gfid);
-			if (rc != UNIFYCR_SUCCESS) {
-				debug("Failed to generate fid for file %s\n", path);
-				return UNIFYCR_ERR_IO;
-			}
-
-			gfid = abs(gfid);
-
-			unifycr_fattr_t *ptr_meta = NULL;
-			rc = get_global_file_meta(gfid, &ptr_meta);
-			if (ptr_meta == NULL) {
-				fid = -1;
-			} else {
-				/* other process has created this file, but its
-				 * attribute is not cached locally,
-				 * allocate a file id slot for this existing file */
-				fid = unifycr_fid_create_file(path);
-				if (fid < 0){
-				   debug("Failed to create new file %s\n", path);
-				   return UNIFYCR_ERR_NFILE;
-				}
-
-				/* initialize the storage for the file */
-				int store_rc = unifycr_fid_store_alloc(fid);
-				if (store_rc != UNIFYCR_SUCCESS) {
-					debug("Failed to create storage for file %s\n", path);
-					return UNIFYCR_ERR_IO;
-				}
-				/* initialize the global metadata
-				 * */
-				unifycr_filemeta_t* meta = unifycr_get_meta_from_fid(fid);
-				meta->real_size = ptr_meta->file_attr.st_size;
-				ptr_meta->fid = fid;
-				ptr_meta->gfid = gfid;
-
-				ins_file_meta(&unifycr_fattrs,\
-						ptr_meta);
-				free(ptr_meta);
-				}
-			} else {
-
-			}
-		}
+        if (fid < 0) {
+            rc = unifycr_get_global_fid(path, &gfid);
+            if (rc != UNIFYCR_SUCCESS) {
+                debug("Failed to generate fid for file %s\n", path);
+                return UNIFYCR_ERR_IO;
+            }
+
+            gfid = abs(gfid);
+
+            unifycr_fattr_t *ptr_meta = NULL;
+            rc = get_global_file_meta(gfid, &ptr_meta);
+            if (ptr_meta == NULL) {
+                fid = -1;
+            } else {
+                /* other process has created this file, but its
+                 * attribute is not cached locally,
+                 * allocate a file id slot for this existing file */
+                fid = unifycr_fid_create_file(path);
+                if (fid < 0) {
+                    debug("Failed to create new file %s\n", path);
+                    return UNIFYCR_ERR_NFILE;
+                }
+
+                /* initialize the storage for the file */
+                int store_rc = unifycr_fid_store_alloc(fid);
+                if (store_rc != UNIFYCR_SUCCESS) {
+                    debug("Failed to create storage for file %s\n", path);
+                    return UNIFYCR_ERR_IO;
+                }
+                /* initialize the global metadata
+                 * */
+                unifycr_filemeta_t *meta = unifycr_get_meta_from_fid(fid);
+                meta->real_size = ptr_meta->file_attr.st_size;
+                ptr_meta->fid = fid;
+                ptr_meta->gfid = gfid;
+
+                ins_file_meta(&unifycr_fattrs,
+                              ptr_meta);
+                free(ptr_meta);
```

Comment:
Not for this PR, but may want to check return statements above for possible memory leak.